manta-crypto/src/merkle_tree/forest.rs

```diff
@@ -593,6 +593,60 @@ macro_rules! impl_from_items_and_witnesses {
                 result
             }
         }
+
+        impl<C, const N: usize> BatchInsertion for $forest
+        where
+            C: Configuration + ?Sized,
+            C::Index: FixedIndex<N>,
+            Parameters<C>: Clone,
+            Leaf<C>: Clone,
+            LeafDigest<C>: Clone + Default + PartialEq,
+            InnerDigest<C>: Clone + Default + PartialEq,
+        {
+            /// # Note
+            ///
+            /// This implementation does not yield the same result as individually inserting
+            /// each item in `items` with [`insert`](Accumulator::insert) till failure.
```

Comment:
Why wouldn't it yield the same result? Each tree is independent, so it shouldn't matter what order you compute these in.

Reply:
It doesn't yield the same result as the default impl of `BatchInsertion` for `Forest`, which would stop after the first failed insertion (even though there may be trees with empty slots left). This custom implementation only halts the insertions tree-wise, which makes more sense. I'm only adding the comment because the implementation doesn't 100% adhere to the comment in the `BatchInsertion` trait, but it is strictly better.